### DISCUSSION TITLE
Name-able references

### DISCUSSION THREAD

**Discussion by zamu-flowerpot** [loco-rs/loco#770] [General]
One thing I think that is currently not properly taken care of is the fact that all references have to be named as a singular table name and it expects the plural for that table to just be a tacked on 's'. 

This is not great since there are plenty of words that pluralize differently; additionally, what happens when there is multiple references to user in a single table. 

First example that comes to mind a Ticketing system where you have a reporter and a assignee for the ticket. I know I could just work around this by making Reporter and Assignee tables which point to user, but that feels wrong and very inefficient from a DB perspective. 

Maybe I've just gotten it wrong or I misunderstood how to use the `generate model` subcommand? 

If this is just missing functionality, then I definitely recommend something like `assignee:references(user)` as the type declaration.

**Comment by jondot** [comment]
Hi, inflection (converting singular/plurals, etc) can indeed be made better. If you can point out the specific places you bumped into a simplistic "tacked-on" 's', we could address those (please feel free to open an issue). We've initially started with simplistic inflection, well, for the sake of keeping things simple (and avoiding staying off-focus with "which kinds of english inflection we want" eg. people vs persons).

Regarding references, maybe you mean the using the *same* table for a reference and discriminating by a type column?

**Comment by aaronpeacock** [comment]
I am not Zamu-Flowerpot (OP)
but it seems to me like they were describing a simple issue-tracker,  where they  clearly wanted to have a "tickets" table with multiple foreign-keys to "users" table,  one of which could be the "issue reporter" and one could be the "assigned person to resolve it"
This requires the column name, as well as the table name being referenced.

This was resolved by PR 955 which you merged earlier today, in my opinion.

I too was waiting for this :D

I would mark this as resolved by PR 955

**Reply by zamu-flowerpot** [reply]
Yep, this resolved the base issue I had with the CLI excluding the misery that is making plurals work.

**Comment by sanderjson** [comment]
I am a bit confused with the correct use, and I believe I found conflicting information.

In general should I use `cargo loco generate scaffold posts` or `cargo loco generate scaffold post`?

Also, RE: inflections — this [Rails reference](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/inflections.rb) might help.
